### DISCUSSION TITLE
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/glmark2-es2 Error: main: Could not initialize canvas

### DISCUSSION THREAD

**Discussion by PugalendhiMurugan** [OE4T/meta-tegra-community#115] [Q&A]
Could anyone please check and let me know, whether you are able to execute glmark2-es2 from the above path? Getting errored as below: *********
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/glmark2-es2 Error: main: Could not initialize canvas 
*********
I'm in need of testing agx orin borad GPU feature, Is there any script or cli to test GPU feature?
Kindly do share me, that would really help me on preparing test script for testing agx orin borad GPU feature?

**Comment by madisongh** [comment]
We do functional testing using the [graphics demo sample programs](https://github.com/OE4T/meta-tegra/blob/master/recipes-graphics/l4t-graphics-demos/l4t-graphics-demos_35.5.0.bb) from NVIDIA.

I've never tried any of the GL benchmarking tools in the meta-openembedded repo, so I don't know if they'll work without modification when built for the Jetsons, but you should make sure that you're running them from an X11 terminal window on your target system, so NVIDIA's GL library can find the display.  You can't run them from a serial console or ssh session.  I'd recommend using the `demo-image-sato` recipe from our [demo distro](https://github.com/OE4T/tegra-demo-distro) as a starting point.

**Reply by PugalendhiMurugan** [reply]
Sure, I will give a try as you said.
Thank you @madisongh
